src/FactSystem/FactMetaData.cc

```diff
@@ -875,13 +875,32 @@ void FactMetaData::_setAppSettingsTranslators(void)
     if (!_enumStrings.count() && (type() == valueTypeDouble || type() == valueTypeFloat)) {
         for (size_t i=0; i<sizeof(_rgAppSettingsTranslations)/sizeof(_rgAppSettingsTranslations[0]); i++) {
             const AppSettingsTranslation_s* pAppSettingsTranslation = &_rgAppSettingsTranslations[i];
-            if (pAppSettingsTranslation->rawUnits == _rawUnits.toLower() && (
-                (pAppSettingsTranslation->unitType == UnitTemperature && pAppSettingsTranslation->unitOption == qgcApp()->toolbox()->settingsManager()->unitsSettings()->temperatureUnits()->rawValue().toUInt()) ||
-                (pAppSettingsTranslation->unitType == UnitSpeed       && pAppSettingsTranslation->unitOption == qgcApp()->toolbox()->settingsManager()->unitsSettings()->speedUnits()->rawValue().toUInt()) ||
-                (pAppSettingsTranslation->unitType == UnitDistance    && pAppSettingsTranslation->unitOption == qgcApp()->toolbox()->settingsManager()->unitsSettings()->distanceUnits()->rawValue().toUInt()) ||
-                (pAppSettingsTranslation->unitType == UnitArea        && pAppSettingsTranslation->unitOption == qgcApp()->toolbox()->settingsManager()->unitsSettings()->areaUnits()->rawValue().toUInt())))
-            {
-                _cookedUnits = pAppSettingsTranslation->cookedUnits;
+
+            if (!(_rawUnits == pAppSettingsTranslation->rawUnits) && !(_rawUnits.toLower() == pAppSettingsTranslation->rawUnits)) {
+                continue;
+            }
+
+            uint settingsUnits = 0;
+
+            switch (pAppSettingsTranslation->unitType) {
+            case UnitDistance:
+                settingsUnits = qgcApp()->toolbox()->settingsManager()->unitsSettings()->distanceUnits()->rawValue().toUInt();
```

Comment:
While cleaning this up would also be good to put ```qgcApp()->toolbox()->settingsManager()->unitsSettings()``` into a variable and using that instead of repeating all of it the over and over and over again.